### ISSUE TITLE
more permissive unregister access

### ISSUE THREAD

**Issue by graybeal** [mmisw/orr-portal#149]
@carueda suggests that we might loosen up the unpublish ontology' access:

> Enable the Unregister option also for the owner of the ontology but only when the visibility of the ontology is owner and perhaps also when the status is not stable

I think for owner-private ontologies they should be able to unregister whatever they want. There is no concern in that case about the ontology being depended on by other people.

If the ontology is public, I hypothesize that any non-stable ontology (draft, unstable, untested) could be retracted. Yes, it may be that some people would have used those ontologies, but there is an "at your own risk" aspect to that, it seems to me.  A thought in that case is that the "Unpublish" command trigger "Are you sure?" guidance that explains why it may be worth leaving the ontology, e.g.:

> We ask that before unregistering your ontology you consider whether anyone else might already be using it, and whether it could offer future value to other users even though it is no longer useful to you. 

The possible responses could be "Good point, cancel my request" and "I still want to unregister my ontology".

Also, creating a 'deprecated' status might make people more comfortable leaving their ontology online. Would it be hard to add that? Would I need to create a separate ticket?

**Comment by carueda** [comment]
Just for reference:

The confirmation dialog looks like this when about to unregister a particular version:

> ![image](https://user-images.githubusercontent.com/556505/66146678-1c234380-e5c2-11e9-94b5-4f213bc37c60.png)

and like this when about to unregister the complete entry for an ontology (including all revisions):

> ![image](https://user-images.githubusercontent.com/556505/66146747-3826e500-e5c2-11e9-893b-7778858630e2.png)

**Comment by lewismc** [comment]
Nice job @carueda

**Comment by carueda** [comment]
ha, I haven't done anything yet `;)` ... just thinking about adding more info to these dialogs along the lines of what John is suggesting above..

**Comment by carueda** [comment]
@graybeal @lewismc 

Toward implementing this, here is how these dialogs look like now in my local dev environment:

For a particular version:

>  ![image](https://user-images.githubusercontent.com/556505/66158096-0c175e00-e5da-11e9-8258-af7789ef3715.png)


For a whole ontology entry:

>  ![image](https://user-images.githubusercontent.com/556505/66158039-e9854500-e5d9-11e9-9bfe-b13ca5658666.png)

Note that I have included more details about the affected version(s) so the user have more context before proceeding with the unregistration.

For easier implementation, without compromising too much the functionality/usability (IMO), I'm already allowing the owner (whether direct user or on behalf of submitting organization) to proceed with the unregistration.  Adding more rules to this logic would take more effort (and time is very limited).

If you agree with this approach (which I've just implemented), I can include this as part of the next 3.8.8 release (along with the also recently implemented #150).
